lib/modulesync/git.rb

```diff
@@ -19,16 +19,18 @@ def self.pull(git_base, name, branch, opts)
 
       # Repo already cloned, check out master and override local changes
       else
-        puts "Overriding any local changes to repositories in #{PROJ_ROOT}"
-        repo = ::Git.open("#{PROJ_ROOT}/#{name}")
-        repo.branch(branch).checkout
-        repo.reset_hard
-        repo.pull('origin', branch)
+        if ! git_provider.eql? 'gerrit'
```

Comment:
The local repo should start with an updated state regardless of the git provider.